pootle/core/checks/checker.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import logging
+import time
+
+from django.conf import settings
+from django.utils import timezone
+from django.utils.functional import cached_property
+from django.utils.lru_cache import lru_cache
+
+from pootle_misc.checks import run_given_filters
+from pootle_misc.util import import_func
+from pootle_store.models import QualityCheck, Unit, Store
+from pootle_store.unit import UnitProxy
+from pootle_translationproject.models import TranslationProject
+
+
+logger = logging.getLogger(__name__)
+
+
+class CheckableUnit(UnitProxy):
+
+    def __init__(self, unit):
+        self.unit = unit
+
+    @property
+    def store(self):
+        return self.store__id
+
+    @property
+    def tp(self):
+        return self.store__translation_project__id
+
+
+class UnitQualityCheck(object):
+
+    def __init__(self, checker, unit, original_checks,
+                 check_names, keep_false_positives=True):
+        """Refreshes QualityChecks for a Unit
+
+        :param checker: a Checker for this Unit.
+        :param unit: an instance of Unit.
+        :param original_checks: current QCs for this Unit
```

Comment:
`QCs`? Please use the long version since this is meant to be a human readable text.